src/main/fc/init.c

```diff
@@ -278,15 +278,15 @@ void init(void)
     // initialize IO (needed for all IO operations)
     IOInitGlobal();
 
-#ifdef USE_HARDWARE_REVISION_DETECTION
-    detectHardwareRevision();
-#endif
-
 #if defined(USE_TARGET_CONFIG)
     // Call once before the config is loaded for any target specific configuration required to support loading the config
     targetConfiguration();
 #endif
 
+#if defined(USE_CONFIG_TARGET_PREINIT)
```

Comment:
another option is weak function with a default noop implementation, then the define here would not be needed. But the define in the config does make it more obvious that this feature is used

Reply:
Agree. The definition makes it very explicit.